packages/ui5-task-stringreplacer/lib/stringreplacer.js

```diff
@@ -0,0 +1,80 @@
+const replaceStream = require("replacestream");
+const dotenv = require("dotenv");
+const log = require("@ui5/logger").getLogger(
+  "builder:customtask:stringreplacer"
+);
+
+dotenv.config();
+
+// get all environment variables
+const envVariables = process.env;
+let placeholderStrings = [];
+
+// loop through env variables to find keys which are having prefix 'stringreplacer'
+if (typeof envVariables === "object") {
+  for (key in envVariables) {
+    // env variable should start with 'stringreplacer' and should in format 'stringreplacer.placeholder'
+    if (key.startsWith("stringreplacer")) {
+      if (!key.split(".")[1]) {
+        log.error(
+          "Failed to replace strings. Expected format is stringreplacer.placeholder=value"
+        );
+      }
+      let placeholderString = key.split(".")[1];
```

Comment:
does this work for multi-"dotted" keys à la `stringreplacer.some.deeply.nested.key = value` as well?

Reply:
As per current logic, nested keys wont work. Whatever present immediately after stringreplacer prefix will be considered as key. 

From your example `stringreplacer.some.deeply.nested.key`, word `some` will be considered as KEY.

Reply:
With current version we should be able to provide deeply nested keys.